composer/trainer/trainer.py

```diff
@@ -2373,7 +2366,7 @@ def save_checkpoint(self, name: str = 'ep{epoch}-ba{batch}-rank{rank}', *, weigh
             weights_only (bool, optional): See :func:`.save_checkpoint`.
 
         Returns:
-            List[pathlib.Path]: See :func:`.save_checkpoint`.
+            str or None: See :func:`.save_checkpoint`.
```

Comment:
is it me or is this function confusingly named given that there is already a `save_checkpoint` and a `_save_checkpoint`? It looks recursive at first glance. any way to change the naming here without messing up the interface?

Reply:
Big +1. This is confusing

Reply:
This is here because we wanted an easy way to for the user to save their checkpoints directly from the trainer,e.g.
```
trainer.save_checkpoint('my_checkpoint.pt')
```
instead of:
```
from composer.utils.checkpoint import save_checkpoint
save_checkpoint(trainer.state, 'my_checkpoint.pt')
```
which is a lot of unnecessary boilerplate. The user also has to remember where that utility function is being imported from everytime.